dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/delete-payment/delete-payment.event-handler.test.ts

```diff
@@ -326,4 +327,136 @@ describe('handleUtilisationReportPaymentDeletedEvent', () => {
       }),
     );
   });
+
+  it('should set hasCorrections to true when the fee record has corrections', async () => {
+    // Arrange
+    const requestSource = aDbRequestSource();
+    const paymentId = 123;
+
+    const report = UtilisationReportEntityMockBuilder.forStatus(RECONCILIATION_IN_PROGRESS).build();
+
+    const linkedFeeRecordEntity = FeeRecordEntityMockBuilder.forReport(report).withId(456).build();
+    const correctionEntity = FeeRecordCorrectionEntityMockBuilder.forFeeRecordAndIsCompleted(linkedFeeRecordEntity, true).build();
+    linkedFeeRecordEntity.corrections = [correctionEntity];
+
+    const paymentEntity = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withId(paymentId).withFeeRecords([linkedFeeRecordEntity]).build();
+    when(mockFindOne).calledWith(PaymentEntity, expect.anything()).mockResolvedValue(paymentEntity);
+    when(mockFindOne).calledWith(FeeRecordEntity, expect.anything()).mockResolvedValue(linkedFeeRecordEntity);
```

Comment:
can something more specific be used instead of expect.anything?

Reply:
I'm not sure I see the value? We have assertions that test that it is called with the correct parameters.

We have to use 'when' here because we need to have separate mocks for when 'findOne' is called to find a payment and called to find a fee record, and this is the minimal way to set up that mocking. 

It would be a lot of additional code to maintain if the specifics of the query changed - and we do have assertions for the specific query which would be getting updated.